### PR TITLE
test(replays): Add timers to recording processing

### PR DESCRIPTION
**Pull request by cmanallen**
For canary use only.

closes: https://github.com/getsentry/replay-backend/issues/252

#skip-changelog

### CODE REVIEW

**Comment by cmanallen** [comment]
@jjbayer @jan-auer Requires approval by someone on the ingest team.  Do you want to deploy this tomorrow morning and we can follow up async with results?

**Comment by cmanallen** [comment]
Also I couldn't use the timers directly in the relay-replays module because of cross-dependency issues.  So I had to make a lot of stuff public and refactor the API a bit.  It should all work the same as before but the behavior is duplicated in the processor so we can time everything.

**Comment by cmanallen** [comment]
Also @jan-auer and @jjbayer if you think this is ready to go feel free to merge it tomorrow and we can look over the results when I wake up.  Otherwise I'll try to be on early to handle peer review feedback.

File: relay-server/src/actors/processor.rs [hunk 1]
```diff
@@ -1124,6 +1125,50 @@ impl EnvelopeProcessorService {
...
+        let mut events = metric!(timer(RelayTimers::ReplayRecordingProcessing), {
+            recording::deserialize(buffer)?
+        });
+
+        metric!(timer(RelayTimers::ReplayRecordingProcessing), {
+            recording::strip_pii(&mut events)
+                .map_err(recording::RecordingParseError::ProcessingAction)?
+        });
+
+        let bytes = metric!(timer(RelayTimers::ReplayRecordingProcessing), {
```
Comment (JoshFerge) [inline relay-server/src/actors/processor.rs]: do we have to create a unique timer for each thing we process to get the differentiated metrics? or does it use the module name or something?

Comment (cmanallen) [reply relay-server/src/actors/processor.rs]: Oversight.  Nice catch.

File: relay-server/src/actors/processor.rs [hunk 2]
```diff
@@ -1124,6 +1125,50 @@ impl EnvelopeProcessorService {
...
+            recording::strip_pii(&mut events)
+                .map_err(recording::RecordingParseError::ProcessingAction)?
+        });
+
+        let bytes = metric!(timer(RelayTimers::ReplayRecordingSerialize), {
+            recording::serialize(events)?
+        });
+        let compressed_bytes = metric!(timer(RelayTimers::ReplayRecordingCompress), {
+            recording::compress(bytes)?
+        });
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs:1183]: Should we also add histogram metrics for the original compressed and decompressed size, + processed compressed and decompressed size?

Usage example: https://github.com/getsentry/relay/blob/1bf9c64d85a1a3d91c57d6f9fab77aed551e904d/relay-server/src/body/store_body.rs#L46

**Review by jjbayer — COMMENTED** [review]
I would not merge this branch to master. Rather remove the condition on the organization ID and then deploy the branch directly to canary.

**Review by JoshFerge — COMMENTED** [review]
based on our discussion yesterday, i'm not quite sure that this will help us solve the problem, which is finding the source of the increased memory usage. does ingest have thoughts on metrics we can add to help with the telemetry of this?

**Comment by jjbayer** [comment]
> based on our discussion yesterday, i'm not quite sure that this will help us solve the problem, which is finding the source of the increased memory usage. does ingest have thoughts on metrics we can add to help with the telemetry of this?

@JoshFerge I would at least add some metrics for the size of the buffers involved (see comment above).

**Pull request closed without merging**